pkg/store/catalog.go

```diff
@@ -124,3 +148,109 @@ func (c *Catalog) GetPlanOf(serviceInstanceSpec *sc_v1b1.ServiceInstanceSpec) (*
 		return nil, errors.Errorf("informer reported multiple instances for ClusterServicePlan %q", planKey)
 	}
 }
+
+func makePlanSchemaKey(plan *sc_v1b1.ClusterServicePlan, action planSchemaAction) planSchemaKey {
+	return planSchemaKey(fmt.Sprintf("%s/%s/%s", plan.Name, action))
+}
+
+func (c *Catalog) getSchemaCache(plan *sc_v1b1.ClusterServicePlan, action planSchemaAction) (*gojsonschema.Schema, bool) {
+	key := makePlanSchemaKey(plan, action)
+
+	c.schemasRWMutex.RLock()
+	defer c.schemasRWMutex.RUnlock()
+	if schemaWithRv, ok := c.schemas[key]; ok && schemaWithRv.resourceVersion == plan.ResourceVersion {
+		return schemaWithRv.schema, true
```

Comment:
If the version stored is old we should remove it. Also as a separate PR later we can remove invalidated schemas on update events for plans - the ones that have different resource versions.

Reply:
I don't think this is necessary - it's safer/easier to reason about if we just 'remove' it by replacing it once we have the new value (see below). Simpler logic, and less risk of unnecessary thrashing around if weird resource processing order.

Reply:
Makes sense - we only have a read lock here.